### PR TITLE
Allow null for `nl2br`

### PR DESCRIPTION
**Pull request by ruudk**
Same as https://github.com/twigphp/Twig/pull/3617
Fixes `Passing null to parameter #1 ($string) of type string is deprecated` on PHP 8.1

### CODE REVIEW

**Comment by ruudk** [comment]
@fabpot ready for review. The failing test is not related.

File: tests/Fixtures/filters/nl2br.test [hunk 1]
```diff
@@ -3,6 +3,7 @@
 --TEMPLATE--
 {{ "I like Twig.\nYou will like it too.\n\nEverybody like it!"|nl2br }}
 {{ text|nl2br }}
+{{ null|nl2br }}
```
Comment (fabpot) [inline tests/Fixtures/filters/nl2br.test]:
```suggestion
*{{ null|nl2br }}*
```

So that we have `**` in the output (like I did in the referenced PR)

File: tests/IntegrationTest.php [hunk 2]
```diff
@@ -229,7 +229,7 @@ public function nl2br($value, $sep = '<br />')
     {
         // not secure if $value contains html tags (not only entities)
         // don't use
-        return str_replace("\n", "$sep\n", $value);
+        return str_replace("\n", "$sep\n", $value ?? '');
```
Comment (fabpot) [inline tests/IntegrationTest.php:232]: Why is it needed?

File: tests/Fixtures/filters/nl2br.test (hunk 1 above)
Comment (fabpot) [reply tests/Fixtures/filters/nl2br.test]: Can you also add a test with an empty string? (again, for consistency with the referenced PR)

File: src/Extension/CoreExtension.php [hunk 3]
```diff
@@ -231,7 +231,7 @@ public function getFilters()
             new TwigFilter('lower', 'twig_lower_filter', ['needs_environment' => true]),
             new TwigFilter('striptags', 'strip_tags'),
```
Comment (fabpot) [inline src/Extension/CoreExtension.php]: I suppose we have the same issue with this filter? And maybe all the other ones that call PHP's native function directly with no wrappers?

**Review by fabpot — CHANGES_REQUESTED** [review]

File: tests/IntegrationTest.php (hunk 2 above)
Comment (ruudk) [reply tests/IntegrationTest.php:232]: Because of this:
```

  1x: str_replace(): Passing null to parameter #3 ($subject) of type array|string is deprecated
    1x in IntegrationTest::testIntegration from Twig\Tests
```

File: tests/Fixtures/filters/nl2br.test (hunk 1 above)
Comment (ruudk) [reply tests/Fixtures/filters/nl2br.test]: Done

File: src/Extension/CoreExtension.php (hunk 3 above)
Comment (ruudk) [reply src/Extension/CoreExtension.php]: I'll add the following:
* `sprintf`
* `strip_tags`

Should we also do something with `abs`? When null, pass 0?

**Comment by ruudk** [comment]
@fabpot Updated the PR, added missing test for striptags filter, and made sure that striptags and format filters now also accept null inputs.

File: src/Extension/CoreExtension.php [hunk 4]
```diff
@@ -481,6 +481,19 @@ function twig_date_modify_filter(Environment $env, $date, $modifier)
     return $date->modify($modifier);
 }
 
+/**
+ * Return a formatted string.
```
Comment (fabpot) [inline src/Extension/CoreExtension.php]:
```suggestion
 * Returns a formatted string.
```

File: src/Extension/CoreExtension.php [hunk 5]
```diff
@@ -1080,6 +1105,19 @@ function twig_lower_filter(Environment $env, $string)
     return mb_strtolower($string ?? '', $env->getCharset());
 }
 
+/**
+ * Strip HTML and PHP tags from a string.
```
Comment (fabpot) [inline src/Extension/CoreExtension.php]:
```suggestion
 * Strips HTML and PHP tags from a string.
```

**Review by fabpot — CHANGES_REQUESTED** [review]

**Comment by ruudk** [comment]
Fixed

**Comment by fabpot** [comment]
Thank you @ruudk.
